### ISSUE TITLE
Cpu usage peaks

### ISSUE THREAD

**Issue by katsar0v** [jupyterlab/jupyterlab#3825]
**Problem**:

Short cpu usage peak when visiting ```http://localhost:8888/lab```

**How I produced it**:

I have a conda environment with jupyter lab installed and 4 local jupyter lab extensions which I installed via ```jupyter labextension install .```. After I installed them and run ```jupyter lab``` the Lab opens in my browser and there is a short cpu usage peak which can be noticed via ```top```. 

**Notes**:

* Is this getting worse with each local extension?
* Why is jupyter lab building extensions every time I visit ```http://localhost:8888/lab``` and refresh it? This causes short cpu peaks and I am not sure whether this is really needed

**Comment by blink1073** [comment]
Hi @katsar0v, the CPU usage you are seeing comes from the fact that we look at locally installed extensions to make sure they have not changed on-disk (requiring a re-build).  Extensions that are installed from npm or a tarball are not looked at.

**Comment by blink1073** [comment]
Yes, this check occurs at page load time.  We could optimize this to happen only the first time you launch the server or give an option to disable the check entirely.

**Comment by katsar0v** [comment]
That would be great if this could happen only if jupyterlab loads for the first time or if there is any config available. Is there something I could do to prevent this, like install from a tarball?

**Comment by blink1073** [comment]
Yes, you could run `npm pack` on your local folder, and then `jupyter labextension install` that tarball.

**Comment by katsar0v** [comment]
Thank you!

**Comment by blink1073** [comment]
Cheers!
